File: miscel/mpi/exercise/ex1a_solution.f90
program pcolor 

implicit none

include "mpif.h"

integer :: nprocs,ierr,myid

! Initialize MPI
call mpi_init(ierr)

call mpi_comm_size(mpi_comm_world,nprocs,ierr)
call mpi_comm_rank(mpi_comm_world,myid,ierr)

if (mod(myid,2).eq.0) then
   write(*,'("Process",I3," has the color red.")') myid
else
   write(*,'("Process",I3," has the color green.")') myid
endif

call mpi_finalize(ierr)

stop
end program pcolor

File: miscel/mpi/ex1a_solution.c
#include <stdio.h>
#include <stdlib.h>
#include "mpi.h"

int main(int argc, char* argv[])
{

  // Initialize MPI.

  MPI_Init(&argc,&argv);
  int nprocs, myid;
  MPI_Comm_size(MPI_COMM_WORLD, &nprocs);
  MPI_Comm_rank(MPI_COMM_WORLD, &myid);

  // Print out to screen.

  if (myid%2 == 0) 
    printf("Process %d has the color red.\n",myid);
  else
    printf("Process %d has the color green.\n",myid);

  // Finalize MPI.

  MPI_Finalize();

  return 0;

}
